pytorch_pfn_extras/training/extension.py

```diff
@@ -222,3 +224,52 @@ def decorator(ext: 'ExtensionLike') -> 'ExtensionLike':
 
 def _as_extension(ext: 'ExtensionLike') -> Extension:
     return ext if isinstance(ext, Extension) else _WrappedExtension(ext)
+
+
+class ExtensionEntry:
+    """Extension and options.
+
+    Args:
+        extension: An extension.
+        name: Name of extension. If omitted, inferred from the attributes of
+            ``extension`` argument.
+        priority: Invocation priority of the extension.
+            Default is ``PRIORITY_READER``.
+        trigger: Trigger object that determines when to invoke the extension.
+            Default is ``(1, 'iteration')``.
+        call_before_training: Flag to call extension before training.
+            Default is ``False``.
+
+    .. seealso::
+       :meth:`pytorch_pfn_extras.training.ExtensionsManager.extend`
+    """
+
+    def __init__(
+            self,
+            extension: 'ExtensionLike',
+            *,
+            name: Optional[str] = None,
+            priority: int = Extension.priority,
+            trigger: 'TriggerLike' = Extension.trigger,
```

Comment:
The default value of `priority` and `trigger` needs to be provided by an extension, like `name`.